src/simplification/Simplifier.ts

```diff
@@ -73,57 +48,80 @@ export class Simplifier {
       }
 
       const simplifiedItems = simplifyItems(schema, this);
-      if (simplifiedItems.newModels !== undefined) {
-        models = [...models, ...simplifiedItems.newModels];
-      }
-      if (simplifiedItems.items !== undefined) {
-        model.items = simplifiedItems.items;
+      if (simplifiedItems !== undefined) {
+        model.items = simplifiedItems;
       }
 
       const simplifiedProperties = simplifyProperties(schema, this);
-      if (simplifiedProperties.properties !== undefined) {
-        model.properties = simplifiedProperties.properties;
-      }
-      if (simplifiedProperties.newModels !== undefined) {
-        models = [...models, ...simplifiedProperties.newModels];
+      if (simplifiedProperties !== undefined) {
+        model.properties = simplifiedProperties;
       }
       
       const simplifiedAdditionalProperties = simplifyAdditionalProperties(schema, this, model);
-      if (simplifiedAdditionalProperties.newModels !== undefined) {
-        models = [...models, ...simplifiedAdditionalProperties.newModels];
-      }
-      if (simplifiedAdditionalProperties.additionalProperties !== undefined) {
-        model.additionalProperties = simplifiedAdditionalProperties.additionalProperties;
+      if (simplifiedAdditionalProperties !== undefined) {
+        model.additionalProperties = simplifiedAdditionalProperties;
       }
 
-      if (this.options.allowInheritance) {
-        const simplifiedExtends = simplifyExtend(schema, this);
-        if (simplifiedExtends.newModels !== undefined) {
-          models = [...models, ...simplifiedExtends.newModels];
-        }
-        if (simplifiedExtends.extendingSchemas !== undefined) {
-          model.extend = simplifiedExtends.extendingSchemas;
-        }
+      const simplifiedExtends = simplifyExtend(schema, this);
+      if (simplifiedExtends !== undefined) {
+        model.extend = simplifiedExtends;
       }
 
       const enums = simplifyEnums(schema);
       if (enums !== undefined && enums.length > 0) {
-        if (model.enum) {
-          model.enum = [...model.enum, ...enums];
-        } else {
-          model.enum = enums;
-        }
+        model.enum = enums;
       }
 
       const required = simplifyRequired(schema);
       if (required !== undefined) {
         model.required = required;
       }
     }
+    this.ensureModelsAreSplit(model);
+    const modelsToReturn = Object.values(this.iteratedModels);
+    //Add models which have not been iterated before
+    if (isModelObject(model) && this.iteratedModels[`${model.$id}`] === undefined) {
```

Comment:
For what you use string template, for avoid solar errors?

Reply:
Yes :)